src/OmniSharp.MiscellanousFiles/MiscellanousFiles.cs

```diff
@@ -0,0 +1,132 @@
+
+namespace OmniSharp.MiscellanousFiles
+{
+    [ExtensionOrder(After = ProjectSystemNames.MSBuildProjectSystem)]
+    [ExtensionOrder(After = ProjectSystemNames.DotNetProjectSystem)]
+    [ExportProjectSystem(ProjectSystemNames.MiscellanousFilesProjectSystem), Shared]
+    public class MiscellanousFilesProjectSystem : IProjectSystem
+    {
+        private readonly string miscFileExtension = ".cs";
+        public string Key { get; } = "MiscellanousFiles";
+        public string Language { get; } = LanguageNames.CSharp;
+        IEnumerable<string> IProjectSystem.Extensions => new[] { miscFileExtension };
+        public bool EnabledByDefault { get; } = true;
+
+        private readonly ConcurrentDictionary<string, DocumentId> _documents = new ConcurrentDictionary<string, DocumentId>();
+        private readonly OmniSharpWorkspace _workspace;
+        private readonly IFileSystemWatcher _fileSystemWatcher;
+        private readonly FileSystemHelper _fileSystemHelper;
+        private readonly List<IUpdates> _projectSystems;
+        private readonly ILogger _logger;
+        private ProjectId _projectId;
+
+        [ImportingConstructor]
+        public MiscellanousFilesProjectSystem(OmniSharpWorkspace workspace, IFileSystemWatcher fileSystemWatcher, FileSystemHelper fileSystemHelper,
+            ILoggerFactory loggerFactory, [ImportMany] IEnumerable<Lazy<IProjectSystem, ProjectSystemMetadata>> projectSystems)
+        {
+            _workspace = workspace;
+            _fileSystemWatcher = fileSystemWatcher ?? throw new ArgumentNullException(nameof(fileSystemWatcher));
+            _fileSystemHelper = fileSystemHelper;
+            _logger = loggerFactory.CreateLogger<MiscellanousFilesProjectSystem>();
+            _projectSystems = projectSystems.
+                Where
+                (ps => ps.Metadata.Name == ProjectSystemNames.MSBuildProjectSystem || 
+                ps.Metadata.Name == ProjectSystemNames.DotNetProjectSystem).
+                Select(ps => ps.Value).
+                Select(ps => ps as IUpdates)
+                .ToList();
+        }
+
+        Task<object> IProjectSystem.GetProjectModelAsync(string filePath)
+        {
+            return Task.FromResult<object>(null);
+        }
+
+        Task<object> IProjectSystem.GetWorkspaceModelAsync(WorkspaceInformationRequest request)
+        {
+            return Task.FromResult<object>(null);
+        }
+
+        void IProjectSystem.Initalize(IConfiguration configuration)
+        {
+            var allFiles = _fileSystemHelper.GetFiles("**/*.cs");
+            foreach (var filePath in allFiles)
+                AddIfMiscellanousFile(filePath);
+
+            _fileSystemWatcher.Watch(miscFileExtension, OnMiscellanousFileChanged);
+        }
+
+        private async void AddIfMiscellanousFile(string filePath)
+        {
+            //wait for the project systems to finish processing the updates
+            foreach(var projectSystem in _projectSystems)
```

Comment:
Do we need to wait for the other project systems each time we process a file, or can we do it once outside this method?